### PR TITLE
M0001 verify timeout

### PR DESCRIPTION
**Pull request by emiltin**
Test 'timeout' property of M0001 #142 

Currently the M0001 is only testing that yellow flash and dark mode works. There should also be test cases (1 or 2?) that checks that the "timeout" property works. If this is set to >0 then the TLC should revert to previous function position (always NormalMode?) when this time is reached.

### CODE REVIEW

**Comment by emiltin** [comment]
the gem tlc test fail. it's because the emulator doesn't yet implment the yellow flash timeout - so the test seems to work correctly.

**Comment by mikaelbroms** [comment]
yellow flash can be activated with M0001 and goes back to NormalControl after one minute
I have removed task.sleep(90)
added optional parameter 'timeot' to helper method wait_normal_control 
pass 120 seconds as timeout to method wait_normal_control
The test doesn´t pass.

Error in detailed log: undefined local variable or method `e'
https://github.com/rsmp-nordic/rsmp_validator/runs/5230295998?check_suite_focus=true

**Comment by otterdahl** [comment]
> Error in detailed log: undefined local variable or method `e' https://github.com/rsmp-nordic/rsmp_validator/runs/5230295998?check_suite_focus=true

I've opened an issue at: https://github.com/rsmp-nordic/rsmp/issues/35

**Comment by emiltin** [comment]
> the gem tlc test fail. it's because the emulator doesn't yet implment the yellow flash timeout - so the test seems to work correctly.

the gem tlc now supports yellow flash timeout, and the test pass

**Comment by emiltin** [comment]
rebased on master

File: spec/site/tlc/modes_spec.rb [hunk 1]
```diff
@@ -179,8 +179,7 @@
     Validator::Site.connected do |task,supervisor,site|
       prepare task, site
       switch_yellow_flash 1
-      task.sleep(90)
-      wait_normal_control
+      wait_normal_control 120
```
Comment (emiltin) [inline spec/site/tlc/modes_spec.rb:182]: the yellow flash timeout used in 1 minute, ie. 60 seconds. i guess a little bit of buffer is reasonable, but do we need 30 seconds? i think this extra time should be a test config, not hardcoded

File: spec/site/tlc/modes_spec.rb [hunk 2]
```diff
@@ -107,7 +107,7 @@
     specify 'yellow flash can be activated with M0001', sxl: '>=1.0.7' do |example|
       Validator::Site.connected do |task,supervisor,site|
         prepare task, site
-        switch_yellow_flash
+        switch_yellow_flash 0
```
Comment (emiltin) [inline spec/site/tlc/modes_spec.rb:110]: i would prefer not to ad all these '0' arguments. let's use a defualt argument to avoid them

Comment (emiltin) [reply spec/site/tlc/modes_spec.rb]: i fixed this

**Comment by emiltin** [comment]
the gem tlc now supports m0001 timeout and passes all tests

**Comment by emiltin** [comment]
i added a switch to normal mode at the start of the test. otherwise the test fails if the tlc is not in normal control when the test is run, because the timeout will switch back to something else than normal control.

**Comment by emiltin** [comment]
> Error in detailed log: undefined local variable or method `e'
> https://github.com/rsmp-nordic/rsmp_validator/runs/5230295998?check_suite_focus=true

this was also fixed

**Comment by otterdahl** [comment]
- ITC-3: Fail due to #168
- ITC-2: It works!
- EC-2: Unable to verify

I'm aiming to clean up this PR and merge it.

**Pull request merged**